core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -614,11 +624,25 @@
               (finally
                 (.unlock wm-lock wm-lock-stamp)))))
 
-        (while (.isBlockFull live-chunk)
-          (finish-block! this))
+        (let [force-flush-req (some? @force-flush-tx-id)
```

Comment:
nit: `when-let`?

Reply:
I use the match variable lower down in the `or`'s guarding block/chunk finishing, if you want to provide a rewrite happy to adopt it

Reply:
it's more to cut down on the number of names required - but tbh I shouldn't golf :golf: 

```clojure
(let [force-flush-tx-id @force-flush-tx-id
      force-flush-match (and force-flush-tx-id (= force-flush-tx-id last-tx-id))]
  (cond
    force-flush-match
    (log/info "received flush instruction: no newer transactions found, will flush chunk.")

    force-flush-tx-id
    (log/info "received flush instruction: newer transaction found, so not flushing chunk."))

  ...
)
```